libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/layoutselection/OpenHouseDataLayoutGenerator.java

```diff
@@ -0,0 +1,121 @@
+package com.linkedin.openhouse.datalayout.layoutselection;
+
+import com.linkedin.openhouse.datalayout.config.DataCompactionConfig;
+import com.linkedin.openhouse.datalayout.datasource.FileStat;
+import com.linkedin.openhouse.datalayout.datasource.TableFileStats;
+import java.util.Collections;
+import java.util.List;
+import lombok.Builder;
+import org.apache.spark.api.java.function.FilterFunction;
+import org.apache.spark.api.java.function.MapFunction;
+import org.apache.spark.api.java.function.ReduceFunction;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Encoders;
+
+/**
+ * Data layout optimization strategies generator for OpenHouse. Generates a list of strategies with
+ * cost.
+ */
+@Builder
+public class OpenHouseDataLayoutGenerator implements DataLayoutGenerator {
+  private static final long FILE_BLOCK_SIZE_BYTES = 1024L * 1024L * 256; // 256MB
+  private static final long FILE_BLOCK_MARGIN_BYTES = 1024L * 1024L * 10; // 10MB
+  private static final long NUM_FILE_GROUPS_PER_COMMIT = 50;
+  private static final double FILE_ENTROPY_THRESHOLD = 100.0;
+  private static final long FILE_OPEN_COST_IN_BYTES = 1024L * 1024L * 4; // 4MB
+  private static final long REWRITE_BYTES_PER_SECOND = 1024L * 1024L * 2; // 2MB/s
+  private static final int REWRITE_PARALLELISM = 900; // number of Spark tasks to run in parallel
+  private final TableFileStats tableFileStats;
+
+  /**
+   * Generate a list of data layout optimization strategies based on the table file stats and
+   * historic query patterns.
+   */
+  @Override
+  public List<DataLayoutOptimizationStrategy> generate() {
+    return Collections.singletonList(generateCompactionStrategy());
+  }
+
+  private DataLayoutOptimizationStrategy generateCompactionStrategy() {
+    long totalSizeBytes =
+        tableFileStats
+            .get()
+            .map((MapFunction<FileStat, Long>) FileStat::getSize, Encoders.LONG())
+            .reduce((ReduceFunction<Long>) Long::sum);
+
+    DataCompactionConfig.DataCompactionConfigBuilder configBuilder = DataCompactionConfig.builder();
+
+    long targetBytesSize = calculateTargetBytesSize();
```

Comment:
Looks like a constant value from implementation of the method, do we need a method or constant?